### ISSUE TITLE
Fluxlisting can't regsiter ( Flutter_user )

### ISSUE THREAD

**Issue by ton2engin** [inspireui/support#3528]
**_Step 1 (require): describe detail issues & screenshots_**
+ Detail Issues: 
+ Product & version:
+ Flutter (or React Native) version: 
+ Testing Device/Simulator: 
+ Screenshot issues (drag the file to attach here): 

Hi,sir
Fluxlisting can't regsiter ( Flutter_user ) with refer photo as below ( Use new version ) 
<a href="https://www.img.in.th/image/ULw7rl"><img src="https://www.img.in.th/images/97fde4bf2ce5e9a803d39ceb546c0dab.md.png" alt="97fde4bf2ce5e9a803d39ceb546c0dab.png" border="0" /></a>

Please support this case. 
Thank you.

**_Step 2 (require): submit proof of purchasing the license on http://verify.inspireui.com_**

**_Important Note:_**
- We will close the ticket if missing the proof of purchase on step 2. The screenshot purchase attaches here will be invalid.
- Kindly create only *One Ticket* & includes all the issues, that would help us focus to resolve it better. 
- If your all ticket was resolved & closed, but want to create a new ticket, just do simple step by linking to the previous verified ID (you don't need to submit the form again), for example, #Ticket-ID

Thank you so much for your time 😊

**Comment by philipvu30** [comment]
Hi,

Have you updated the ListApp Mobile Manager to the latest version?

**Comment by ton2engin** [comment]
> Hi,
> 
> Have you updated the ListApp Mobile Manager to the latest version?

Hi,
Yes, it lastes version is 1.6.3 .

**Comment by philipvu30** [comment]
Hi,

Would you mind uploading the woo_commerce.dart file to here? It's in lib/services/woo_commerce.dart.

Thanks.

**Comment by ton2engin** [comment]
> Hi,
> 
> Would you mind uploading the woo_commerce.dart file to here? It's in lib/services/woo_commerce.dart.
> 
> Thanks.

Hi,

Please check file = https://drive.google.com/file/d/1MY1rEENllhW46I69q6Z7XOtMeBkPfYGl/view?usp=sharing

Thank you.

**Comment by philipvu30** [comment]
Hi @ton2engin ,

Please approve the email: vutha******30@gmail.com.

Thank you.

**Comment by philipvu30** [comment]
Hi,

It was a mistake that we uploaded the wrong version of ListApp Mobile Manager plugin.
Please manually delete the plugin on your website and upload this [plugin](https://github.com/inspireui/support/files/4722691/listapp-mobile-manager.zip) instead. 

Thank you for your patience.

**Comment by ton2engin** [comment]
> website

Hi,
- I manually delete the plugin on website and upload your plugin and install already, after that i test register is fail with same last problem Flutter_user 
- I already approve the email: vutha******30@gmail.com.

Thank you.

**Comment by philipvu30** [comment]
Hi,

Please replace these files in lib/services. Thank you
[SERVICES.zip](https://github.com/inspireui/support/files/4727259/SERVICES.zip)

**Comment by ton2engin** [comment]
> Hi,
> 
> Please replace these files in lib/services. Thank you
> [SERVICES.zip](https://github.com/inspireui/support/files/4727259/SERVICES.zip)

Hi sir, 
After replace,It build fail with as below 

`Launching lib/main.dart on iPhone SE (2nd generation) in debug mode...

Compiler message:
lib/services/listing.dart:266:26: Error: The getter 'status' isn't defined for the class 'Review'.
 - 'Review' is from 'package:fstore/models/review.dart' ('lib/models/review.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'status'.
              if (review.status == 'approved') {
                         ^^^^^^
Running pod install...                                             46.6s
Running Xcode build...
Xcode build done.                                           465.5s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    In file included from /Users/mactj/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.m:1:
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:4:4: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    + (instancetype)sharedInstance;
       ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:4:4: note: insert '_Nullable' if the pointer may be null
    + (instancetype)sharedInstance;
       ^
                    _Nullable 
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:4:4: note: insert '_Nonnull' if the pointer should never be null
    + (instancetype)sharedInstance;
       ^
                    _Nonnull 
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:7:33: warning: block pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
          restorationHandler:(void (^)(NSArray *_Nullable))restorationHandler;
                                    ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:7:33: note: insert '_Nullable' if the block pointer may be null
          restorationHandler:(void (^)(NSArray *_Nullable))restorationHandler;
                                    ^
                                     _Nullable
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:7:33: note: insert '_Nonnull' if the block pointer should never be null
          restorationHandler:(void (^)(NSArray *_Nullable))restorationHandler;
                                    ^
                                     _Nonnull
    2 warnings generated.
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:15:30: warning: 'requestReview' is only available on iOS 10.3 or newer [-Wunguarded-availability]
        [SKStoreReviewController requestReview];
                                 ^~~~~~~~~~~~~~
    In module 'StoreKit' imported from /Users/mactj/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:2:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/StoreKit.framework/Headers/SKStoreReviewController.h:22:1: note: 'requestReview' has been marked as being introduced in iOS 10.3 here, but the deployment target is iOS 8.0.0
    + (void)requestReview API_AVAILABLE(ios(10.3), macos(10.14)) API_UNAVAILABLE(watchos) __TVOS_PROHIBITED;
    ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:15:30: note: enclose 'requestReview' in an @available check to silence this warning
        [SKStoreReviewController requestReview];
                                 ^~~~~~~~~~~~~~
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:15:6: warning: 'SKStoreReviewController' is only available on iOS 10.3 or newer [-Wunguarded-availability]
        [SKStoreReviewController requestReview];
         ^~~~~~~~~~~~~~~~~~~~~~~
    In module 'StoreKit' imported from /Users/mactj/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:2:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/StoreKit.framework/Headers/SKStoreReviewController.h:14:110: note: 'SKStoreReviewController' has been marked as being introduced in iOS 10.3 here, but the deployment target is iOS 8.0.0
    SK_EXTERN_CLASS API_AVAILABLE(ios(10.3), macos(10.14)) API_UNAVAILABLE(watchos) __TVOS_PROHIBITED @interface SKStoreReviewController : NSObject
                                                                                                                 ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:15:6: note: enclose 'SKStoreReviewController' in an @available check to silence this warning
        [SKStoreReviewController requestReview];
         ^~~~~~~~~~~~~~~~~~~~~~~
    2 warnings generated.
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:196:43: warning: 'FIRMessagingRemoteMessage' is deprecated: FCM direct channel is deprecated, please use APNs for downstream message handling. [-Wdeprecated-declarations]
    - (void)applicationReceivedRemoteMessage:(FIRMessagingRemoteMessage *)remoteMessage {
                                              ^
    In module 'FirebaseMessaging' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:219:1: note: 'FIRMessagingRemoteMessage' has been explicitly marked deprecated here
    __deprecated_msg(
    ^
    In module 'UIKit' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/sys/cdefs.h:191:48: note: expanded from macro '__deprecated_msg'
            #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                          ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:308:24: warning: 'FIRMessagingRemoteMessage' is deprecated: FCM direct channel is deprecated, please use APNs for downstream message handling. [-Wdeprecated-declarations]
        didReceiveMessage:(FIRMessagingRemoteMessage *)remoteMessage {
                           ^
    In module 'FirebaseMessaging' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:219:1: note: 'FIRMessagingRemoteMessage' has been explicitly marked deprecated here
    __deprecated_msg(
    ^
    In module 'UIKit' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/sys/cdefs.h:191:48: note: expanded from macro '__deprecated_msg'
            #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                          ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:142:30: warning: 'shouldEstablishDirectChannel' is deprecated: FCM direct channel is deprecated, please use APNs channel for downstream message delivery. [-Wdeprecated-declarations]
        [FIRMessaging messaging].shouldEstablishDirectChannel = true;
                                 ^
    In module 'FirebaseMessaging' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:283:56: note: 'shouldEstablishDirectChannel' has been explicitly marked deprecated here
    @property(nonatomic) BOOL shouldEstablishDirectChannel DEPRECATED_MSG_ATTRIBUTE(
                                                           ^
    In module 'UIKit' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/AvailabilityMacros.h:182:64: note: expanded from macro 'DEPRECATED_MSG_ATTRIBUTE'
                #define DEPRECATED_MSG_ATTRIBUTE(s) __attribute__((deprecated(s)))
                                                                   ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:197:52: warning: 'appData' is deprecated [-Wdeprecated-declarations]
      [self didReceiveRemoteNotification:remoteMessage.appData];
                                                       ^
    In module 'FirebaseMessaging' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:226:62: note: 'appData' has been explicitly marked deprecated here
    @property(nonatomic, readonly, strong) NSDictionary *appData DEPRECATED_ATTRIBUTE;
                                                                 ^
    In module 'UIKit' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/AvailabilityMacros.h:180:60: note: expanded from macro 'DEPRECATED_ATTRIBUTE'
            #define DEPRECATED_ATTRIBUTE        __attribute__((deprecated))
                                                               ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:307:1: warning: implementing deprecated method [-Wdeprecated-implementations]
    - (void)messaging:(FIRMessaging *)messaging
    ^
    In module 'FirebaseMessaging' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:253:1: note: method 'messaging:didReceiveMessage:' declared here
    - (void)messaging:(FIRMessaging *)messaging
    ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:309:63: warning: 'appData' is deprecated [-Wdeprecated-declarations]
      [_channel invokeMethod:@"onMessage" arguments:remoteMessage.appData];
                                                                  ^
    In module 'FirebaseMessaging' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:226:62: note: 'appData' has been explicitly marked deprecated here
    @property(nonatomic, readonly, strong) NSDictionary *appData DEPRECATED_ATTRIBUTE;
                                                                 ^
    In module 'UIKit' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/AvailabilityMacros.h:180:60: note: expanded from macro 'DEPRECATED_ATTRIBUTE'
            #define DEPRECATED_ATTRIBUTE        __attribute__((deprecated))
                                                               ^
    6 warnings generated.

    Compiler message:
    lib/services/listing.dart:266:26: Error: The getter 'status' isn't defined for the class 'Review'.
     - 'Review' is from 'package:fstore/models/review.dart' ('lib/models/review.dart').
    Try correcting the name to the name of an existing getter, or defining a getter or field named 'status'.
                  if (review.status == 'approved') {
                             ^^^^^^
    Target kernel_snapshot failed: Exception: Errors during snapshot creation: null
    build failed.
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: Mapping architecture armv7 to i386. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'image_picker' from project 'Pods')
    warning: Mapping architecture arm64 to x86_64. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'image_picker' from project 'Pods')

Could not build the application for the simulator.
Error launching application on iPhone SE (2nd generation).
`

Thank you.

**Comment by philipvu30** [comment]
Hi,

I've sent you the latest project via your email. Please check.

**Comment by ton2engin** [comment]
Hi sir,

**After i replace file have effect as below** 
- I use new version FluxStore-listing 1.4.1 ,Why need replace new version again ?

`Launching lib/main.dart on iPhone SE (2nd generation) in debug mode...
Running Xcode build...

Compiler message:
lib/widgets/home/banner/banner_map.dart:44:23: Error: A value of type 'PermissionStatus' can't be assigned to a variable of type 'bool'.
 - 'PermissionStatus' is from 'package:location/location.dart' ('../../.pub-cache/hosted/pub.dartlang.org/location-2.5.4/lib/location.dart').
        _permission = await _locationService.requestPermission();
                      ^
Xcode build done.                                           52.9s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    In file included from /Users/mactj/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.m:1:
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:4:4: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    + (instancetype)sharedInstance;
       ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:4:4: note: insert '_Nullable' if the pointer may be null
    + (instancetype)sharedInstance;
       ^
                    _Nullable 
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:4:4: note: insert '_Nonnull' if the pointer should never be null
    + (instancetype)sharedInstance;
       ^
                    _Nonnull 
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:7:33: warning: block pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
          restorationHandler:(void (^)(NSArray *_Nullable))restorationHandler;
                                    ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:7:33: note: insert '_Nullable' if the block pointer may be null
          restorationHandler:(void (^)(NSArray *_Nullable))restorationHandler;
                                    ^
                                     _Nullable
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:7:33: note: insert '_Nonnull' if the block pointer should never be null
          restorationHandler:(void (^)(NSArray *_Nullable))restorationHandler;
                                    ^
                                     _Nonnull
    2 warnings generated.
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:15:30: warning: 'requestReview' is only available on iOS 10.3 or newer [-Wunguarded-availability]
        [SKStoreReviewController requestReview];
                                 ^~~~~~~~~~~~~~
    In module 'StoreKit' imported from /Users/mactj/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:2:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/StoreKit.framework/Headers/SKStoreReviewController.h:22:1: note: 'requestReview' has been marked as being introduced in iOS 10.3 here, but the deployment target is iOS 8.0.0
    + (void)requestReview API_AVAILABLE(ios(10.3), macos(10.14)) API_UNAVAILABLE(watchos) __TVOS_PROHIBITED;
    ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:15:30: note: enclose 'requestReview' in an @available check to silence this warning
        [SKStoreReviewController requestReview];
                                 ^~~~~~~~~~~~~~
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:15:6: warning: 'SKStoreReviewController' is only available on iOS 10.3 or newer [-Wunguarded-availability]
        [SKStoreReviewController requestReview];
         ^~~~~~~~~~~~~~~~~~~~~~~
    In module 'StoreKit' imported from /Users/mactj/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:2:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/StoreKit.framework/Headers/SKStoreReviewController.h:14:110: note: 'SKStoreReviewController' has been marked as being introduced in iOS 10.3 here, but the deployment target is iOS 8.0.0
    SK_EXTERN_CLASS API_AVAILABLE(ios(10.3), macos(10.14)) API_UNAVAILABLE(watchos) __TVOS_PROHIBITED @interface SKStoreReviewController : NSObject
                                                                                                                 ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:15:6: note: enclose 'SKStoreReviewController' in an @available check to silence this warning
        [SKStoreReviewController requestReview];
         ^~~~~~~~~~~~~~~~~~~~~~~
    2 warnings generated.
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:196:43: warning: 'FIRMessagingRemoteMessage' is deprecated: FCM direct channel is deprecated, please use APNs for downstream message handling. [-Wdeprecated-declarations]
    - (void)applicationReceivedRemoteMessage:(FIRMessagingRemoteMessage *)remoteMessage {
                                              ^
    In module 'FirebaseMessaging' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:219:1: note: 'FIRMessagingRemoteMessage' has been explicitly marked deprecated here
    __deprecated_msg(
    ^
    In module 'UIKit' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/sys/cdefs.h:191:48: note: expanded from macro '__deprecated_msg'
            #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                          ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:308:24: warning: 'FIRMessagingRemoteMessage' is deprecated: FCM direct channel is deprecated, please use APNs for downstream message handling. [-Wdeprecated-declarations]
        didReceiveMessage:(FIRMessagingRemoteMessage *)remoteMessage {
                           ^
    In module 'FirebaseMessaging' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:219:1: note: 'FIRMessagingRemoteMessage' has been explicitly marked deprecated here
    __deprecated_msg(
    ^
    In module 'UIKit' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/sys/cdefs.h:191:48: note: expanded from macro '__deprecated_msg'
            #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                          ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:142:30: warning: 'shouldEstablishDirectChannel' is deprecated: FCM direct channel is deprecated, please use APNs channel for downstream message delivery. [-Wdeprecated-declarations]
        [FIRMessaging messaging].shouldEstablishDirectChannel = true;
                                 ^
    In module 'FirebaseMessaging' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:283:56: note: 'shouldEstablishDirectChannel' has been explicitly marked deprecated here
    @property(nonatomic) BOOL shouldEstablishDirectChannel DEPRECATED_MSG_ATTRIBUTE(
                                                           ^
    In module 'UIKit' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/AvailabilityMacros.h:182:64: note: expanded from macro 'DEPRECATED_MSG_ATTRIBUTE'
                #define DEPRECATED_MSG_ATTRIBUTE(s) __attribute__((deprecated(s)))
                                                                   ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:197:52: warning: 'appData' is deprecated [-Wdeprecated-declarations]
      [self didReceiveRemoteNotification:remoteMessage.appData];
                                                       ^
    In module 'FirebaseMessaging' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:226:62: note: 'appData' has been explicitly marked deprecated here
    @property(nonatomic, readonly, strong) NSDictionary *appData DEPRECATED_ATTRIBUTE;
                                                                 ^
    In module 'UIKit' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/AvailabilityMacros.h:180:60: note: expanded from macro 'DEPRECATED_ATTRIBUTE'
            #define DEPRECATED_ATTRIBUTE        __attribute__((deprecated))
                                                               ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:307:1: warning: implementing deprecated method [-Wdeprecated-implementations]
    - (void)messaging:(FIRMessaging *)messaging
    ^
    In module 'FirebaseMessaging' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:253:1: note: method 'messaging:didReceiveMessage:' declared here
    - (void)messaging:(FIRMessaging *)messaging
    ^
    /Users/mactj/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:309:63: warning: 'appData' is deprecated [-Wdeprecated-declarations]
      [_channel invokeMethod:@"onMessage" arguments:remoteMessage.appData];
                                                                  ^
    In module 'FirebaseMessaging' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:226:62: note: 'appData' has been explicitly marked deprecated here
    @property(nonatomic, readonly, strong) NSDictionary *appData DEPRECATED_ATTRIBUTE;
                                                                 ^
    In module 'UIKit' imported from /Users/mactj/AndroidStudioProjects/sengran4/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/AvailabilityMacros.h:180:60: note: expanded from macro 'DEPRECATED_ATTRIBUTE'
            #define DEPRECATED_ATTRIBUTE        __attribute__((deprecated))
                                                               ^
    6 warnings generated.

    Compiler message:
    lib/widgets/home/banner/banner_map.dart:44:23: Error: A value of type 'PermissionStatus' can't be assigned to a variable of type 'bool'.
     - 'PermissionStatus' is from 'package:location/location.dart' ('../../.pub-cache/hosted/pub.dartlang.org/location-2.5.4/lib/location.dart').
            _permission = await _locationService.requestPermission();
                          ^
    Target kernel_snapshot failed: Exception: Errors during snapshot creation: null
    build failed.
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: Mapping architecture armv7 to i386. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'image_picker' from project 'Pods')
    warning: Mapping architecture arm64 to x86_64. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'image_picker' from project 'Pods')

Could not build the application for the simulator.
Error launching application on iPhone SE (2nd generation).
`

**Comment by michaelbrinkworth** [comment]
https://www.facebook.com/groups/289288585778355/

**Comment by philipvu30** [comment]
Hi @ton2engin 

Because the file I gave you is the latest, and version 1.4.1 had some issues that we did not aware of.

**Comment by ton2engin** [comment]
Hi Sir,

OK, I understand .
Now, I can logging as below
- IOS = Facebook, Google >>> OK , Register (Flutter_user) >> can't 
- Android = Facebook >>> OK , Register(Flutter_user)  , google ( Sign_in_fail )  >> Can't 

So  what should i do or waiting update ( Another have this problem or not ? )
Thank you for support.

**Comment by philipvu30** [comment]
Hi,

On Android, google firebase must be configured correctly. Please check that if you have added your SHA1 key to Firebase.

And for the "register user" function, it seems one of your plugins or your configuration has prevented the access from outside.

![image](https://user-images.githubusercontent.com/54442328/83989888-0694c380-a972-11ea-862a-a23b1a0b331c.png)

**Comment by ton2engin** [comment]
Hi sir,

I think, i mistake to use Debug Keystore for Firebase ,But i can't get SHA1 from FluxListing.
I try to use in https://medium.com/@inspireui/setup-google-login-for-flutter-apps-fedc58a490b8 but can't get SHA1. ( This method for Mstore only ? becasue Fluxlisting not have mstore.keystore )
Note: i check password,keyalias,storeFile in key.properties 

I try to use => keytool -exportcert -list -v -alias key -keystore android/certs/key.jks
but show => keytool error: java.lang.Exception: Only one command is allowed: both -exportcert and -list were specified.

Thank you.

**Comment by philipvu30** [comment]
Hi,

You can get the SHA 1 with this easy solution. 
1. Right-click the android folder
2. Open in terminal
3. Run this command: ./gradlew signingReport

You will be able to get the Debug SHA1 key. Please note that this key is for testing only. It means only the google account associated with your created Firebase can log in.

To get the Release keys, please refer to this official [document](https://developers.google.com/android/guides/client-auth) 

Thank you.

**Comment by philipvu30** [comment]
Closed as no response in 2 days. 

If you are happy with our services/support, would you mind rating us on your [download page](https://codecanyon.net/downloads)?
We truly appreciate your help.

**Comment by ton2engin** [comment]
Hi sir, 

OK,Now i can Register&login with Facebook and Google already, but can't register becasue Flutter_user Error from my picture.

So,I think to request user login with Facebook and Google are OK.
Thank you for support.

**Comment by philipvu30** [comment]
Hi,

Please check that if you have this function coded in lib/services/woo_commerce.dart and **make sure the URL is the same as the picture**

![image](https://user-images.githubusercontent.com/54442328/84462304-e67a4280-ac98-11ea-9f53-d63c77ce8ffb.png)

**Comment by ton2engin** [comment]
Hi sir,
OK, I test with change URL after that can register but can't login.
My register request = First name, Last name, Enter your Email, Enter your password 
My Login request = Username, Password 
So i try to use E-mail and First name for login but show error as below 

<img width="323" alt="Screen Shot 2563-06-12 at 23 05 49" src="https://user-images.githubusercontent.com/65043311/84523065-e4989980-ad01-11ea-8515-eeff676b0dc2.png">

Thank you so much.

**Comment by philipvu30** [comment]
Hi,

That username should be the email you registered.

**Comment by philipvu30** [comment]
Closed as no response in 2 days. 

If you are happy with our services/support, would you mind rating us on your [download page](https://codecanyon.net/downloads)?
We truly appreciate your help.
